agent/cli/ehrql_telemetry.py

```diff
@@ -0,0 +1,124 @@
+import argparse
+import json
+import os
+import sys
+from datetime import datetime
+from pathlib import Path
+
+from opentelemetry import context, trace
+
+from agent.executors.local import job_metadata_path
+from common.tracing import setup_default_tracing
+
+
+def main(argv):
+    parser = argparse.ArgumentParser(description="db_telemetry")
+    parser.add_argument("dataset", help="Honeycomb dataset to emit telemetry to")
+    parser.add_argument(
+        "job", help="job to emit telemetry for. Either job id or path to job log dir."
+    )
+    parser.add_argument(
+        "logfile",
+        nargs="?",
+        type=Path,
+        help="ehrql json log file (defaults to logs.json in the log dir)",
+    )
+
+    args = parser.parse_args(argv)
+
+    if os.path.isdir(args.job):
+        metadata_path = Path(args.job) / "metadata.json"
+    else:
+        metadata_path = job_metadata_path(args.job)
+    metadata = json.loads(metadata_path.read_text())
+
+    assert "ehrql" in metadata["container_metadata"]["Config"]["Image"], (
+        "Cowardly refusing to emit telemetry for non-ehrql job"
+    )
+
+    if args.logfile:
+        logfile_path = args.logfile
+    else:
+        logfile_path = metadata_path.parent / "logs.json"
```

Comment:
I feel like we could make this significantly easier to use if we got it to do the conversion itself. I don't think it's much more complicated than:
```py
def parse_ehrql_logs(logs_text):
    process = subprocess.run(
        ["docker", "run" "--rm", "ghcr.io/opensafely-core/ehrql:v1" "/app/scripts/parse_logs.py"],
        input=text,
        encoding="utf-8",
        capture_output=True,
        check=True,
    )
    return process.stdout
```

Obviously I'll need to get my changes to ehrQL merged, and you might want to stub it out in tests, but feels like it could be a big usability win.

Reply:
done